kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/BaseContentService.java

```diff
@@ -13431,6 +13454,13 @@ public void setOldDisplayName(String oldDisplayName) {
 		public void open();
 
 		/**
+		 * Get a direct link to the asset so it doesn't have to be streamed.
+		 * @param resource
+		 * @return URI or null if no direct link is available
+		 */
+		public URI getDirectLink(ContentResource resource);
```

Comment:
Doesn't allow it to work on earlier JVMs but I wonder for public APIs like this if we should start returning `Optional<URI>` to make it clearer the caller should check to see if there is a value present.